### PR TITLE
[SELS-66] [FE] Login

### PR DESCRIPTION
**Pull request by presyos**
# Related Links
Asana:  https://app.asana.com/0/1203098252776160/1203290430207962/f

# What do you change at this PR?
frontend
add loading component after login and manipulation between disabling the input fields and button

# Target pages
FE
`/`

# Test View Points
1. After inputting the credentials you must see the loading icon if success automatic navigate to the dashboard else toast error will display and the loading icon disappear

# TODO / Things to do after merging:
NA

# Commands to run
Frontend

```
cd frontend
cd e-learning-fe
npm start
```
Backend

```
cd backend
.\venv\Scripts\Activate
python manage.py runserver
```

# ScreenShots
NOTE: This picture is to show the loading icon without restriction for visuality
image

### CODE REVIEW

**Review by Alvilb — COMMENTED** [review]
disable the fields and buttons when the page is loading

File: frontend/e-learning-fe/src/components/login.js [hunk 1]
```diff
@@ -66,12 +70,30 @@ function Login() {
...
+              <path
+                d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
+                fill="currentColor"
+              />
+              <path
+                d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
+                fill="currentFill"
+              />
+            </svg>
+          </div>
```
Comment (Alvilb) [inline frontend/e-learning-fe/src/components/login.js:90]: create a component for loader

**Comment by Alvilb** [comment]
please change how the code is written
```
<button
  type="submit"
  class="w-full text-white bg-primary-600 hover:bg-primary-700 focus:ring-4 ..."
  onClick={handleLogin}
  disabled={isDisable}
>
  Sign in
</button>
```

File: frontend/e-learning-fe/src/components/login.js [hunk 2]
```diff
@@ -27,6 +34,9 @@ function Login() {
         navigate("/admin-dashboard", { replace: true });
       } else navigate("/dashboard", { replace: true });
     } else {
+      setIsLoading(false);
+      setIsDisableField(false);
```
Comment (Alvilb) [inline frontend/e-learning-fe/src/components/login.js:38]: cant we just use `isLoading` for the fields that use `isDisableField` ?

File: frontend/e-learning-fe/src/components/login.js [hunk 3]
```diff
@@ -120,37 +126,29 @@ function Login() {
...
-              >
-                Sign in
-              </button>
-            )}
+            <button
+              type="submit"
+              class={
+                isDisableSignInButton
+                  ? "w-full text-white bg-primary-600 hover:bg-primary-700 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-gray-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
+                  : "w-full text-white bg-primary-600 hover:bg-primary-700 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
```
Comment (Alvilb) [inline frontend/e-learning-fe/src/components/login.js:138]: only these part here is different and the rest is the same right?
disabled: `dark:bg-gray-600`
enabled: `dark:bg-blue-600`

do we have a specific reason why we repeated the parts that are the same?

Comment (presyos) [reply frontend/e-learning-fe/src/components/login.js:138]: Opo pero parang as one kasi sya heheh di ko po sila mahiwalay

Comment (Alvilb) [reply frontend/e-learning-fe/src/components/login.js:138]: sorry last confirmation, this does not work? does it raise an error po ba? or just straight up not working pag ganito :bow: 
```
className = {`w-full text-white bg-primary-600 hover:bg-primary-700 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:hover:bg-primary-700 dark:focus:ring-primary-800 ${isDisabled ? 'dark:bg-blue-600' : 'dark:bg-gray-600'}`}
```

Comment (presyos) [reply frontend/e-learning-fe/src/components/login.js:138]: Working na sya sir mali pala yung ginawa ko, ngayon ko lang po nalaman yang ginawa mo hehehe push ko na po saglit after testing

Comment (presyos) [reply frontend/e-learning-fe/src/components/login.js]: Done sir

**Review by Alvilb — APPROVED** [review]

**Pull request merged**